File: backend-server/src/main/java/com/video/server/entity/UserBehavior.java
package com.video.server.entity;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * 用户行为实体类
 */
@Data
public class UserBehavior {
    
    /**
     * 主键ID
     */
    private Long id;
    
    /**
     * 用户ID
     */
    private Long userId;
    
    /**
     * 视频ID
     */
    private Long videoId;
    
    /**
     * 行为类型（like, view, collect, share等）
     */
    private String actionType;
    
    /**
     * 创建时间
     */
    private LocalDateTime createTime;
}

File: bigdata-engine/src/main/scala/com/shortvideo/recommendation/common/entity/UserBehavior.scala
package com.shortvideo.recommendation.common.entity

import java.sql.Timestamp

/**
 * 用户行为实体类
 * @param userId 用户ID
 * @param videoId 视频ID
 * @param actionType 行为类型
 * @param createTime 创建时间
 * @param duration 观看时长（秒）
 * @param extraInfo 扩展信息
 */
case class UserBehavior(
                         userId: Long,
                         videoId: Long,
                         actionType: String,
                         createTime: Timestamp,
                         duration: Int = 0,
                         extraInfo: String = ""
                       )

/**
 * 用户行为评分实体（用于ALS模型）
 * @param userId 用户ID
 * @param videoId 视频ID
 * @param rating 评分
 * @param timestamp 时间戳
 */
case class UserRating(
                       userId: Long,
                       videoId: Long,
                       rating: Double,
                       timestamp: Long
                     )

/**
 * Kafka中的用户行为消息
 * @param logId 日志ID
 * @param userId 用户ID
 * @param videoId 视频ID
 * @param behavior 行为类型
 * @param timestamp 时间戳
 * @param properties 额外属性
 */
case class UserBehaviorEvent(
                              logId: String,
                              userId: Long,
                              videoId: Long,
                              behavior: String,
                              timestamp: Long,
                              properties: Map[String, String] = Map.empty
                            )